File: backend/app/models/skill.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID
from enum import Enum


class SkillLevel(str, Enum):
    BEGINNER = "BEGINNER"
    INTERMEDIATE = "INTERMEDIATE"
    ADVANCED = "ADVANCED"
    EXPERT = "EXPERT"


class SkillCategory(str, Enum):
    TECHNOLOGY = "Technology"
    BUSINESS = "Business"
    CREATIVE = "Creative"
    LANGUAGE = "Language"
    OTHER = "Other"


class SkillBase(BaseModel):
    name: str
    category: SkillCategory
    level: SkillLevel
    description: Optional[str] = None
    estimated_hours: int = 1


class SkillCreate(SkillBase):
    pass


class Skill(SkillBase):
    id: UUID
    valuation_score: float = 0.0
    created_at: datetime
    
    class Config:
        from_attributes = True


class UserSkillOffered(BaseModel):
    id: UUID
    user_id: UUID
    skill_id: UUID
    proficiency_level: SkillLevel
    years_experience: int = 0
    created_at: datetime
    
    class Config:
        from_attributes = True


class UserSkillRequested(BaseModel):
    id: UUID
    user_id: UUID
    skill_id: UUID
    desired_level: SkillLevel
    priority: int = 1
    created_at: datetime
    
    class Config:
        from_attributes = True

File: core-engine/src/main/java/com/alterix/models/Skill.java
package com.alterix.models;

public class Skill {
    private final String id;
    private String name;
    private String category;
    private SkillLevel level;
    private int estimatedHours;
    private double valuationScore;
    private String description;

    public enum SkillLevel {
        BEGINNER, INTERMEDIATE, ADVANCED, EXPERT
    }

    public Skill(String id, String name, String category, SkillLevel level) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.level = level;
        this.estimatedHours = 0;
        this.valuationScore = 0.0;
    }

    public String getId() { return id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    
    public SkillLevel getLevel() { return level; }
    public void setLevel(SkillLevel level) { this.level = level; }
    
    public int getEstimatedHours() { return estimatedHours; }
    public void setEstimatedHours(int hours) { this.estimatedHours = hours; }
    
    public double getValuationScore() { return valuationScore; }
    public void setValuationScore(double score) { this.valuationScore = score; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    @Override
    public String toString() {
        return "Skill{name='" + name + "', level=" + level + ", value=" + valuationScore + "}";
    }
}
